File: C#/Refactoring/refactoring_exercise_3/za/co/entelect/refactoring3/domain/MoneyMarketAccount.cs
namespace refactoring_exercise_3.za.co.entelect.refactoring3.domain
{

    public class MoneyMarketAccount : BankAccount
    {

        public MoneyMarketAccount(long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) :
            base(balanceInCents, creditInterestsRate, debitInterestRate, accountFee)
        {

        }
        public override AccountType GetAccountType()
        {
            return AccountType.MoneyMarket;
        }
    }
}

File: Java/refactoring_exercise_3/src/main/java/za/co/entelect/refactoring3/domain/MoneyMarketAccount.java
package za.co.entelect.refactoring3.domain;

public class MoneyMarketAccount extends BankAccount {

    public MoneyMarketAccount(Long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) {
        super(balanceInCents, creditInterestsRate, debitInterestRate, accountFee);
    }

    @Override
    public AccountType getAccountType() {
        return AccountType.MONEY_MARKET;
    }
}
